### PR TITLE
sync: Call Destroy instead of Reset

### PR DESCRIPTION
**Pull request by charles-lunarg**
The Destroy call was calling ResetCBState() which results in Reset() being called. Reset adds in a self reference which causes a cycle in the shared_ptr's. Calling Destroy instead cleans everything up properly and doesn't leak any memory.

### CODE REVIEW

File: layers/sync/sync_validation.cpp [hunk 1]
```diff
@@ -4279,7 +4279,7 @@ syncval_state::CommandBuffer::CommandBuffer(SyncValidator *dev, VkCommandBuffer
 syncval_state::CommandBuffer::~CommandBuffer() { Destroy(); }
 
 void syncval_state::CommandBuffer::Destroy() {
-    ResetCBState();  // must be first to clean up self references correctly.
+    access_context.Destroy();  // must be first to clean up self references correctly.
```
Comment (juan-lunarg) [inline layers/sync/sync_validation.cpp:4282]: Looks like a regression caused by me:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/commit/3a7ab5ca48fb9171c22f22fcb832bf4263f6f610

I'd revert the changes made to layers/synchronization_validation.* to make the fix more clear.

Comment (charles-lunarg) [reply layers/sync/sync_validation.cpp:4282]: I can do that, though it does look like some of the changes are needed, like moving `CommandBufferAccessContext access_context;` to be public.

**Review by juan-lunarg — APPROVED** [review]
LGTM

FYI @jeremyg-lunarg for a double check

File: layers/sync/sync_validation.cpp [hunk 2]
```diff
@@ -4276,20 +4276,16 @@ syncval_state::CommandBuffer::CommandBuffer(SyncValidator *dev, VkCommandBuffer
...
     CMD_BUFFER_STATE::Destroy();
 }
 
 void syncval_state::CommandBuffer::Reset() {
     CMD_BUFFER_STATE::Reset();
-    ResetCBState();
+    access_context.Reset();
```
Comment (jeremyg-lunarg) [inline layers/sync/sync_validation.cpp:4286]: 👍 for this

**Review by jeremyg-lunarg — APPROVED** [review]
